### ISSUE TITLE
Modal & Popover take more than 2 seconds to pop up.

### ISSUE THREAD

**Issue by jamesharvey2** [ionic-team/ionic#13747]
Hello guys,

I implemented modal html page and popover component properly on my Ionic app.

I declared them on app.module.ts and on my home.ts:

```
import { NavController, Platform, ModalController, NavParams, PopoverController } from 'ionic-angular';
import { PopoverComponent } from '../../components/popover/popover';
```

in constructor:

```
    public modal: ModalController, 
    private popoverCtrl: PopoverController,

```

and function:
```
openModal() {
  const myModal = this.modal.create('ModalPage');
  myModal.present();
}

presentPopover(myEvent) {
  let popover = this.popoverCtrl.create(PopoverComponent);
  popover.present({
    ev: myEvent
  });
}
```

At beginning of my project, they were not this slow but as my project progress, I learned they're very slow on smartphones. 
On my desktop ionic lab, they respond as they should. They open up in about a second. 
On my smartphone on the other hand, they take more than 2 seconds to show up after click. I added tappable attribute to my (click) on html but they're still too slow to use on any app which intended to behave like a normal app. 

Both modal and popover have nothing on their html content besides a few lines of demo texts. Now they both are unusable. Only thing they will do is give users a bad impression about the app I'm working on. 

Please fix this issue in Ionic 4. Modal & popover should show up in 0.5 second max. 

I'm testing on HTC 710c and Kyocera Reach which are both quad-core CPU android smartphones. Both of them can run 3D games like need for speed and Revolt without slowing down anything. This seems like an issue behind UI programming. I spotted this problem in ion-segment as well... when you have many contents in ion-segment, it performs slower than angular 5 tab with the same contents inside. 

Thanks in advance,

**Comment by kensodemann** [comment]
I am going to close this as something that I cannot duplicate.

I attempted to duplicate using a medium sized application that I commonly use for testing issues. The modal and popover I added contained about 15 widgets each. I tested on an iPhone7, Note 4, and Galaxy Tab S (the Note 4 and Tab S are both older devices). In all cases, the modal and popup were quite responsive. The Tab S had the worst performance, and it still presented the modal and the popover in less than a second.

If you can provide a sample application via GitHub that demonstrates this issue I will be happy to reopen the issue and have a look.

**Comment by jamesharvey2** [comment]
That's so strange... 
I'm using mid-low end smartphones to test this. I will be getting a high end smartphone shortly (LG V10). 

There's nothing wrong with my coding. 
I placed modal & pop over button in a tab menu. 
They take 2+ seconds to show up. 

I will try to make those github pages to demonstrate problems in modal and ion-segment.
Also please test ionic UI on mid-low end smartphones too. 
Kyocera Reach is still a quad core CPU android 5.0 smartphone which can be purchased for $25 used on ebay. HTC 710c runs Android 5.1 for only $65 now. Many people can get what they want from under $100 smartphones which still ship with Android 5.0. 

Here's an example of Alcatel smartphone which ships with Android 6.0 for only $56 price tag.
https://www.ebay.com/itm/Alcatel-PIXI45-5-0-16GB-Android-6-0-Marshmallow-LTE-Unlocked-Smartphones/332443763092?epid=708146277&hash=item4d67314194:g:3ZAAAOSwH2VaBK7y

This still handles 99% of need for people who don't want to spend too much on smartphones. 
I wish you guys consider testing Ionic on those phones too. These phones still can run other commerical game apps without too many problems but one of my testing Alcatel device couldn't run anything built on Cordova. 

There are a plenty of these mid-low smartphones with quad core CPU & Android 5 or 6: 
Blu, Alcatel, TracFone, Leagoo, Kyocera to name a few.

https://www.ebay.com/itm/Leagoo-Kiicaa-Power-5-0-inch-3G-Smartphone-Android-7-0-2BG-16GB-4000mAh-Golden/202151389467?hash=item2f1129691b:g:Gf4AAOSw~QRaN2YY

What they're selling are legitimate smartphones too. None of these were available at this price point in 2007.. but now things have changed and there are more options to buyers.

**Comment by kensodemann** [comment]
I can only test on what I have, which is basically stuff I buy for my own personal use (or in the case of the Note 4, kept as a test machine after it got too old for my personal use). I am not sure what they have at the home office. They probably have some cheapies there. I'll have a look next time I am there (usually about once a week).

**Comment by jamesharvey2** [comment]
Great, investing $45 into those alcatel devices isn't a bad idea because they're now very popular. 
I'm testing everything on low-mid smartphones and can see bugs which I can't observe in other apps built on java or C#.

**Comment by jamesharvey2** [comment]
https://www.kyoceramobile.com/hydro-reach/

https://www.kyoceramobile.com/about/

http://skydevices.com/

**Comment by jamesharvey2** [comment]
Actually, after some research I have found Note 4 has really compelling CPU that is like that of a laptop computer.
It's running quad-core CPU at 2.7Ghz speed. 
`Quad-core 2.7 GHz Krait 450 - Snapdragon 805`
https://www.gsmarena.com/samsung_galaxy_note_4-6434.php

So you're testing everything just like on a laptop.  Samsung Note 4 is from 2014 but it was made to feature laptop-like performance and Samsung nailed it. 

All of these $100-$150 smartphones I have mentioned, no matter how new they are, come with 1.2Ghz to 1.4GHz speed CPU which are much slower than a regular laptop CPUs.  
For example, check this Sony's brand new Xperia L1 from 2017. It costs $120 but it has much inferior CPU than Note 4.

https://www.gsmarena.com/sony_xperia_l1-8619.php

I recommend much inferior testing environment.

**Comment by jamesharvey2** [comment]
https://versus.com/en/intel-core-i7-2670qm-vs-qualcomm-snapdragon-805-apq8084

Here's a comparison between Intel's i7 2nd generation 2670QM laptop CPU and Samsung's sanpdragon 805. Intel created mobile i7 series for workstation laptops. 

When I test Ionic apps on my i7 laptop, I don't see speed problems. ion-segment, modal, popover all work as way they should on Ionic lab browser. I can see these problems on low-mid range smartphones with 1.1Ghz - 1.5Ghz CPUs.

**Comment by kensodemann** [comment]
Here is the cheapest phone I could find in the office: https://www.gsmarena.com/blu_r1_hd-8171.php

I tested on that phone, but saw no speed issues with the popup or the modal. We may have some others around to test on, but that was the lowest end phone I could find.

**Comment by jamesharvey2** [comment]
Thanks for testing. That's strange...because I concluded they're too slow to use on HTC desire smartphone.  I created a hide & show div using angular's *ngif directive with tappable attribute and it's working in a second.    

http://www.angulartutorial.net/2017/03/show-and-hide-in-angular-js-2.html?m=1

I created ion-segment replacement using *ngIf directive with CSS3 and now it's working fine without delay. 


I followed instructions on Josh Morony's website to build modals and pop over. I think I didn't make any mistake. They were too slow to use...ion-segment was responding 1.5+ seconds after I tap.
